kanata: 1.9.0 -> 1.10.0

auscyber opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Updates kanata to 1.10.0
Changelog:
https://github.com/jtroo/kanata/releases/tag/v1.10.0
Has a breaking change the supported karabiner driverkit is now v6
this has been updated in the derivation

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

auscyber committed:
kanata: 1.9.0 -> 1.10.0

Co-authored-by: LIN, Jian <me@linj.tech>

jian-lin reviewed (changes requested):
For darwin, there is a breaking change.  If that is a breaking change for users (I am not a darwin user so I am not sure), we need to:
- postpone the merge according to the 25.11 [release schedule][1]
- mention the breaking change in "Nixpkgs Release Notes" following the PR template
  > Package update: when the change is major or breaking.

[1]: https://www.github.com/NixOS/nixpkgs/issues/443568

auscyber commented:
> For darwin, there is a breaking change. If that is a breaking change for users (I am not a darwin user so I am not sure), we need to:
> 
>     * postpone the merge according to the 25.11 [release schedule](https://www.github.com/NixOS/nixpkgs/issues/443568)
> 
>     * mention the breaking change in "Nixpkgs Release Notes" following the PR template
>       > Package update: when the change is major or breaking.
It is pseudo breaking as long as people use the darwinDriver output it's no difference for the user, if people are not, then yes it is breaking

jian-lin commented:
BTW, have you done any testing?  It is a bit weird to open a PR without ticking any boxes in the PR template.  I can test it on Linux.

auscyber commented:
> BTW, have you done any testing? It is a bit weird to open a PR without ticking any boxes in the PR template. I can test it on Linux.

yeah ive tested it on macos

jian-lin commented:
> It is pseudo breaking as long as people use the darwinDriver output it's no difference for the user, if people are not, then yes it is breaking

Treating it as a real breaking change sounds safe and won't hurt.  Could you add a release note?

jian-lin commented:
I just find the new `updateScript` cannot update `cargoHash`, which makes kanata fail to build. (This PR builds fine.)

jian-lin commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -206,6 +206,8 @@
 
 - `inspircd` has been updated to the v4 release series. Please refer to the upstream documentation for [general information](https://docs.inspircd.org/4/overview/#v4-overview) and a list of [breaking changes](https://docs.inspircd.org/4/breaking-changes/).
 
+- `kanata` now requires karabiner-dk version 6.0+ or later. The package has been updated to use the new karabiner-dk package and the darwinDriver output stays at the version defined in the package.
```
At this time, if we treat this as a normal/real breaking change, it means
- 1.10.0 will not be included in nixos 25.11
- this release note entry should be moved to `rl-2605.section.md` (which will be created on 2025-11-23)

jian-lin committed:
kanata: let updateScript be able to change cargoHash again

This functionality was broken by #445407.  This patch restores it.

jian-lin committed:
kanata: filter versions in updateScript

kanata has pre-release versions like v1.10.0-prerelease-3[1].
updateScript used to be able to filter out those versions but this
functionality was broken by #445407.  This patch restores that
functionality.

[1]: https://github.com/jtroo/kanata/releases/tag/v1.10.0-prerelease-3

jian-lin reviewed (approved):
I moved the release note from 25.11 to 26.05.  I also fixed some `updateScript` issues introduced in #445407.
